svgdigitizer/svgfigure.py

```diff
@@ -0,0 +1,1596 @@
+            ...     <text x="0" y="200">E1: 0 mV</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M 100 200 L 100 100" />
+            ...     <text x="100" y="200">E2: 1 mV</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M -100 100 L 0 100" />
+            ...     <text x="-100" y="100">j1: 0 uA / cm2</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M -100 0 L 0 0" />
+            ...     <text x="-100" y="0">j2: 1 uA / cm2</text>
+            ...   </g>
+            ...   <text x="-200" y="330">scan rate: 50 mV / s</text>
+            ... </svg>'''))
+            >>> figure = SVGFigure(SVGPlot(svg), si_units=True)
+            >>> figure.df
+                  t      E     j
+            0  0.00  0.000  0.00
+            1  0.02  0.001  0.01
+
+        TESTS::
+
+        A dataframe with a time axis, reconstructed with a given scan rate
+        and in SI units, where the y-axis does not have compatible astropy units::
+
+            >>> from svgdigitizer.svg import SVG
+            >>> from svgdigitizer.svgplot import SVGPlot
+            >>> from svgdigitizer.svgfigure import SVGFigure
+            >>> from io import StringIO
+            >>> svg = SVG(StringIO(r'''
+            ... <svg>
+            ...   <g>
+            ...     <path d="M 0 100 L 100 0" />
+            ...     <text x="0" y="0">curve: 0</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M 0 200 L 0 100" />
+            ...     <text x="0" y="200">E1: 0 mV</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M 100 200 L 100 100" />
+            ...     <text x="100" y="200">E2: 1 mV</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M -100 100 L 0 100" />
+            ...     <text x="-100" y="100">j1: 0 persons</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M -100 0 L 0 0" />
+            ...     <text x="-100" y="0">j2: 1 persons</text>
+            ...   </g>
+            ...   <text x="-200" y="330">scan rate: 50 mV / s</text>
+            ... </svg>'''))
+            >>> figure1 = SVGFigure(SVGPlot(svg), si_units=True)
+            >>> figure1.df
+                  t      E    j
+            0  0.00  0.000  0.0
+            1  0.02  0.001  1.0
+
+        """
+        df = self.svgplot.df.copy()
+
+        if self.si_units:
+            for column in df.columns:
+                column_unit = self.figure_schema.get_field(column).custom["unit"]
+                if self.unit_is_astropy(column_unit):
+                    self._convert_axis_to_si(df, column)
+
+        if self.scan_rate:
+            self._add_time_axis(df)
+            return df[["t", self.svgplot.xlabel, self.svgplot.ylabel]]
+
+        return df[[self.svgplot.xlabel, self.svgplot.ylabel]]
+
+    def _convert_axis_to_si(self, df, label):
```

Comment:
from CV module, where specific axes were converted. Nor any kind of axis can be converted.